diff -Naur phpbb3.orig/www/includes/ucp/ucp_profile.php phpbb3/www/includes/ucp/ucp_profile.php
--- phpbb3.orig/www/includes/ucp/ucp_profile.php	2010-03-05 21:37:22.000000000 +0100
+++ phpbb3/www/includes/ucp/ucp_profile.php	2010-11-14 20:19:12.000000000 +0100
@@ -83,6 +83,7 @@
 						$error[] = 'NEW_PASSWORD_ERROR';
 					}
 
+                    /* NIRGAL was here
 					if (($data['new_password'] || ($auth->acl_get('u_chgemail') && $data['email'] != $user->data['user_email']) || ($data['username'] != $user->data['username'] && $auth->acl_get('u_chgname') && $config['allow_namechange'])) && !phpbb_check_hash($data['cur_password'], $user->data['user_password']))
 					{
 						$error[] = 'CUR_PASSWORD_ERROR';
@@ -93,6 +94,8 @@
 					{
 						$error[] = 'SAME_PASSWORD_ERROR';
 					}
+                    */
+                    // END NIRGAL PATCH
 
 					if ($auth->acl_get('u_chgemail') && $data['email'] != $user->data['user_email'] && $data['email_confirm'] != $data['email'])
 					{
@@ -104,6 +107,33 @@
 						$error[] = 'FORM_INVALID';
 					}
 
+                    // NIRGAL WAS HERE
+					if (!sizeof($error)) {
+                        $ch = curl_init('https://'.$user->data['username'].':'.$data['cur_password'].'@'.$_SERVER['HTTP_HOST'].'/hook_change_password');
+                        if ($ch) {
+                            curl_setopt($ch, CURLOPT_POST, TRUE);
+                            curl_setopt($ch, CURLOPT_POSTFIELDS, array('password'=>$data['new_password']));
+                            curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE); //return the transfer as a string of the return instead of outputting it out directly. 
+                            curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, FALSE);
+                            $output = curl_exec($ch);
+                            if (curl_errno($ch))
+                               $error[]="Erreur interne. curl returns ".curl_error($ch);
+                            else {
+                                $http_code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+                                if ($http_code == 401)
+                                    $error[] = 'CUR_PASSWORD_ERROR';
+                                elseif ($http_code != 200)
+                                    $error[]="Erreur interne. Sous-requete HTTP de status $http_code.";
+                                elseif ($output != "OK")
+                                    $error[]=$output;
+                            }
+                            curl_close($ch);
+                        } else {
+                            $error[]='Erreur interne. curl init failed.';
+                        }
+
+                    }
+                    // END NIRGAL PATCH
 					if (!sizeof($error))
 					{
 						$sql_ary = array(
@@ -646,4 +676,4 @@
 	}
 }
 
-?>
\ No newline at end of file
+?>
diff -Naur phpbb3.orig/www/ucp.php phpbb3/www/ucp.php
--- phpbb3.orig/www/ucp.php	2010-03-05 21:37:23.000000000 +0100
+++ phpbb3/www/ucp.php	2010-11-14 20:19:12.000000000 +0100
@@ -87,12 +87,35 @@
 			$user->session_kill();
 			$user->session_begin();
 			$message = $user->lang['LOGOUT_REDIRECT'];
+            // nirgal was here
+            if (strpos($_SERVER["HTTP_USER_AGENT"], 'MSIE')!==FALSE) {
+                $message = '<span style="color:red; font-size:300%;">ERREUR</span><p>
+                Utilisateurs d\'Internet explorer, votre navigateur ne supporte pas la déconnexion!<br>
+                Vous devez fermer toutes les fenêtres d\'Internet Explorer qui sont actuellement ouvertes, y compris sur d\'autres sites.<br>
+                En cas de doute, il est recommandé de fermer purement et simplement votre session, en particulier si vous êtes dans dans un environement peu fiable, type cyber café!';
+            }
+            else {
+                $message = 'Vous allez être déconnecté dans un moment. Veuillez patienter.';
+            }
 		}
 		else
 		{
 			$message = ($user->data['user_id'] == ANONYMOUS) ? $user->lang['LOGOUT_REDIRECT'] : $user->lang['LOGOUT_FAILED'];
 		}
-		meta_refresh(3, append_sid("{$phpbb_root_path}index.$phpEx"));
+        // nirgal was here
+		//meta_refresh(3, append_sid("{$phpbb_root_path}index.$phpEx"));
+        if (strpos($_SERVER["HTTP_USER_AGENT"], 'MSIE')===FALSE) {
+            global $template;
+            if ($_SERVER["HTTPS"])
+                $url = 'https';
+            else
+                $url = 'http';
+            $url.='://logout:logout@'.$_SERVER["HTTP_HOST"].'/logout.php';
+
+            $template->assign_vars(array(
+                'META' => '<meta http-equiv="refresh" content="0;url='.$url.'" />')
+            );
+        }
 
 		$message = $message . '<br /><br />' . sprintf($user->lang['RETURN_INDEX'], '<a href="' . append_sid("{$phpbb_root_path}index.$phpEx") . '">', '</a> ');
 		trigger_error($message);
@@ -346,4 +369,4 @@
 	}
 }
 
-?>
\ No newline at end of file
+?>
